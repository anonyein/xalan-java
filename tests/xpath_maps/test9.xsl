<?xml version="1.0" encoding="UTF-8"?>
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
                xmlns:map="http://www.w3.org/2005/xpath-functions/map"
				exclude-result-prefixes="map"
                version="3.0">
                
   <!-- Author: mukulg@apache.org -->                  			

   <xsl:output method="xml" indent="yes"/>
   
   <!-- An XPath 3.1 test case, to test nesting of arrays and maps within another map. -->

   <xsl:template match="/">
	  <xsl:variable name="map1" select="map {
										    'one' : map {
											           'x1' : 'p',
													   'x2' : 'q',
													   'x3' : 'r',
							                           'x4' : 's',
							                           'x5' : 't' 						   
											        },
											'two' : 2,
											'three' : [5, 6, 7, 8, 9, 'a1'] }"/>
      <xsl:variable name="innerMap" select="map:get($map1, 'one')"/>											
      <innerMap noOfEntries="{count(map:keys($innerMap))}"/>
   </xsl:template>

</xsl:stylesheet>
