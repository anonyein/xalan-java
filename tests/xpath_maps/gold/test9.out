<?xml version="1.0" encoding="UTF-8"?><innerMap>
  <entry>
    <key>x1</key>
    <value>p</value>
  </entry>
  <entry>
    <key>x2</key>
    <value>q</value>
  </entry>
  <entry>
    <key>x3</key>
    <value>r</value>
  </entry>
  <entry>
    <key>x4</key>
    <value>s</value>
  </entry>
  <entry>
    <key>x5</key>
    <value>t</value>
  </entry>
</innerMap>
